controllers/managedocs_controller.go

```diff
@@ -1086,3 +1118,66 @@ func getCSVByPrefix(csvList opv1a1.ClusterServiceVersionList, name string) *opv1
 	}
 	return csv
 }
+
+func (r *ManagedOCSReconciler) reconcileK8sMetricsServiceMonitorAuthSecret() error {
+	r.Log.Info("Reconciling k8s-metrics-service-monitor-auth secret")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.k8sMetricsServiceMonitorAuthSecret, func() error {
+		secret := &corev1.Secret{}
+		secret.Name = grafanaDatasourceSecretName
+		secret.Namespace = openshiftMonitoringNamespace
+		if err := r.unrestrictedGet(secret); err != nil {
+			return fmt.Errorf("Failed to get grafana-datasources secret from openshift-monitoring namespace: %v", err)
+		}
+
+		authInfoStructure := struct {
+			DataSources []struct {
+				BasicAuthPassword string `json:"basicAuthPassword"`
+				BasicAuthUser     string `json:"basicAuthUser"`
+			} `json:"datasources"`
+		}{}
+
+		grafanaDatasourcesPrometheusConfig := secret.Data[grafanaDatasourceSecretKey]
+
+		if err := json.Unmarshal(grafanaDatasourcesPrometheusConfig, &authInfoStructure); err != nil {
```

Comment:
Why create a temp variable for a single access? 

```suggestion
		if err := json.Unmarshal(secret.Data[grafanaDatasourceSecretKey], &authInfoStructure); err != nil {
```